Packages/com.unity.inputsystem/InputSystem/Editor/Dialog.cs

```diff
@@ -0,0 +1,119 @@
+#if UNITY_EDITOR
+
+using System;
+using UnityEditor;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    // Input system UI dialogs centralized as utility methods.
+    // In the future we may introduce possibility to mock/stub these dialogs via delegates to allow
+    // automated testing of dialog options and secure that no dialogs are shown in editor testing.
+    internal static class Dialog
+    {
+        // Represents the result of the user selecting a dialog option.
+        public enum Result
+        {
+            // User decided that unsaved changes should be saved to the associated asset.
+            Save = 0,
+
+            // Operation was cancelled by the user.
+            Cancel = 1,
+
+            // Unsaved changes should be discarded and NOT be saved to the associated asset.
+            Discard = 2,
+
+            // User decided to delete the associated resource.
+            Delete = 3,
+        }
+
+        // User UI dialog windows related to InputActionAssets
+        public static class InputActionAsset
```

Comment:
naming: InputActionAssetDialog?

Reply:
Am I understanding correctly that you would you prefer `Dialog.InputActionAssetDialog.ShowSaveChanges(path)` in favour of `Dialog.InputActionAsset.ShowSaveChanges(path)`?  
Since already prefixed and scope reduced by outer `Dialog` class my intention was to avoid excessive long names and also avoid scope leakage into outer scope which happens when you just use namespace and auto-import. The idea here was to access Dialog since that is what you want to use and then for what kind of type and then use-case. I can change but just want to know if I understand correctly.

Reply:
I think it's good to avoid any confusion with the names in all scenarios, if we are forced to always use Dialog.xx I think this is sufficient.

Reply:
ok will leave it as is for now then and we can reconsider in the future